### DISCUSSION TITLE
How to use health checking with Direct Forwarding?

### DISCUSSION THREAD

**Discussion by hansmbakker** [dotnet/yarp#1867] [Q&A] [answered]
I found the documentation on how to use [health checking with the proxy functionality](https://microsoft.github.io/reverse-proxy/articles/dests-health-checks.html), but what if I use [Direct Forwarding](https://microsoft.github.io/reverse-proxy/articles/direct-forwarding.html)?

**Comment by Tratcher** [comment, marked as answer]
Direct Forwarding is a lower level function and does not have built in healt checks.

You either need to A) do you own health checks, or B) use the higher level model. Why did you choose direct forwarding?

**Reply by hansmbakker** [reply]
> Why did you choose direct forwarding?

This might sound odd, but I need to create an App Service with Easy Auth to add a simple authentication protection layer in front of another application running on another IP. Front Door or Application Gateway do not offer authentication functionality.

However, without an app running inside the App Service, that App Service does nothing. I chose YARP to forward the requests to the backend application.

Since I only needed passthrough for the requests, I thought Direct Forwarding would be the least complex, but I didn't know that it would be more difficult to add functionality like health checks.

**Reply by Tratcher** [reply]
Try the other model and let us know how it works for you.
